packages/strapi-admin/admin/src/containers/Webhooks/EditView/index.js

```diff
@@ -0,0 +1,365 @@
+
+  const formatError = Object.keys(formErrors)
+    .filter(key => key.includes('headers'))
+    .reduce((obj, key) => {
+      obj[key] = formErrors[key];
+      return obj;
+    }, {});
+
+  const headerTitle = isCreating
+    ? formatMessage({
+        id: `Settings.webhooks.create`,
+      })
+    : name;
+
+  const headersActions = [
+    {
+      color: 'primary',
+      disabled: isTriggerActionDisabled,
+      label: formatMessage({
+        id: `Settings.webhooks.trigger`,
+      }),
+      onClick: handleTrigger,
+      style: {
+        padding: '0 15px',
+      },
+      title: isTriggerActionDisabled
+        ? formatMessage({
+            id: `Settings.webhooks.trigger.save`,
+          })
+        : null,
+      type: 'button',
+      icon: (
+        <Play
+          width="14px"
+          height="14px"
+          fill={isTriggerActionDisabled ? '#b4b6ba' : '#ffffff'}
+        />
+      ),
+    },
+    {
+      color: 'cancel',
+      disabled: areActionDisabled,
+      label: formatMessage({
+        id: `app.components.Button.reset`,
+      }),
+      onClick: handleReset,
+      style: {
+        padding: '0 20px',
+      },
+      type: 'button',
+    },
+    {
+      color: 'success',
+      disabled: areActionDisabled,
+      label: formatMessage({
+        id: `app.components.Button.save`,
+      }),
+      style: {
+        minWidth: 140,
+      },
+      type: 'submit',
+    },
+  ];
+
+  const headerProps = {
+    title: {
+      label: headerTitle,
+    },
+    actions: headersActions,
+  };
+
+  const handleBlur = () => {
+    if (submittedOnce) {
+      checkFormErrors();
+    }
+  };
+
+  const handleChange = ({ target: { name, value } }) => {
+    dispatch({
+      type: 'ON_CHANGE',
+      keys: name.split('.'),
+      value,
+    });
+  };
+
+  const handleClick = () => {
+    dispatch({
+      type: 'ADD_NEW_HEADER',
+      keys: ['headers'],
+    });
+  };
+
+  const handleTrigger = async () => {
+    dispatch({
+      type: 'IS_TRIGGERING',
+    });
+
+    try {
+      const { data } = await request(`/admin/webhooks/${id}/trigger`, {
+        method: 'POST',
+        signal,
+      });
+
+      dispatch({
+        type: 'TRIGGER_SUCCEEDED',
+        response: data,
+      });
+    } catch (err) {
+      if (err.code !== 20) {
+        strapi.notification.error('notification.error');
+      }
+      dispatch({
+        type: 'IS_TRIGGERING',
+      });
+    }
+  };
+
+  const handleRemove = index => {
+    dispatch({
+      type: 'ON_HEADER_REMOVE',
+      index,
+    });
+    resetHeadersErrors();
+  };
+
+  const handleReset = () =>
+    dispatch({
+      type: 'RESET_FORM',
+    });
+
+  const handleSubmit = e => {
+    e.preventDefault();
+    setSubmittedOnce(true);
+    checkFormErrors(true);
+  };
+
+  const checkFormErrors = async (submit = false) => {
+    const webhookToCheck = modifiedData;
+    set(webhookToCheck, 'headers', cleanHeaders(modifiedData.headers));
```

Comment:
You don't need these lines c.f. comment in `yup` schema.

Reply:
I need these lines because as we said before, the headers array can be an array with one object with empty values. But we don't want it to create an error, so we have to clean it before trying to validate it. It's also this clean format that will be sent to backend.